docs/workflows/lambda.md

```diff
@@ -6,123 +6,64 @@ hide_title: true
+* [Java 8](https://docs.aws.amazon.com/lambda/latest/dg/java-programming-model.html)
+* [Python 2.7, 3.6](https://docs.aws.amazon.com/lambda/latest/dg/python-programming-model.html)
 
-Modify package.json:
+Addtionally you may chose to include any of the other languages AWS [supports](https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html) as a resource with reduced feature support.
 
-* name
-* version
-* description
-* test script
-* dependencies (NOT devDependencies)
+## Deploy a Lambda
 
-## Build a Lambda
+### Node.js Lambda
 
-To build Node.js Lambda functions, use webpack to pack the Lambda code into a single `.js` file with dependencies:
+For a new Node.js Lambda, create a new function and add an `aws_lambda_function` resource to your Cumulus deployment (for examples, see the example in source [example/lambdas.tf](https://github.com/nasa/cumulus/blob/master/example/cumulus-tf/lambdas.tf) and [ingest/lambda-functions.tf](https://github.com/nasa/cumulus/blob/master/tf-modules/ingest/lambda-functions.tf)) as either a new `.tf` file, or added to an existing `.tf` file:
 
-```bash
-  $ npm run build
-```
+```tcl
+resource "aws_lambda_function" "function" {
+  function_name    = "${var.prefix}-function"
+  filename         = "/path/to/zip/lambda.zip"
+  source_code_hash = filebase64sha256("/path/to/zip/lambda.zip/lambda.zip")
+  handler          = "index.handler"
+  role             = module.cumulus.lambda_processing_role_arn
+  runtime          = "nodejs8.10"
 
-Alternatively, to monitor for changes and auto-rebuild:
+  tags = local.default_tags
 
-```bash
-  $ npm run watch
+  vpc_config {
+    subnet_ids         = var.subnet_ids
+    security_group_ids = var.security_group_ids
+  }
+}
 ```
 
-For non-node Lambdas not included in Cumulus repo, upload .zip to s3 and modify lambdas.yml as previously shown.
+**Please note**: This example contains the minimum set of required configuration.
 
-## Deploy a Lambda
-
-### Node.js Lambda
+Make sure to include a `vpc_config` that matches the information you've provided the `cumulus` module if intending to integrate the lambda with a Cumulus deployment.
 
-For new Node.js Lambdas, update `<daac>-deploy/lambdas.yml` by adding a new entry.
-
-```yaml
-    <LambdaName>:                                   # eg:  LambdaSample (does not need to conform to dirname)
-      handler: <dir>.<function>                     # eg:  sample-lambda.handler (assuming file has module.exports.handler = <someFunc>)
-      timeout: <s>                                  # eg:  300
-      source: 'node_modules/@cumulus/<dir>/dist/'   # eg:  '../cumulus/cumulus/tasks/sample-lambda/dist/index.js'
-```
+Also note that for this example to work, you will need to have `default_tags` defined as in the [example](https://github.com/nasa/cumulus/blob/master/example/cumulus-tf/main.tf), or in the [template-deploy-repo](https://github.com/nasa/cumulus-template-deploy/blob/CUMULUS-1556-add-deployment/cumulus-tf/main.tf).
```

Comment:
Should we mention that you can exclude it if you don't care about tagging? Or conversely, not include tagging in the snippet but then mention how to do it here?

Reply:
I had hoped it'd be implied that if you remove the tags key, it won't add a tag to the lambda resource.     I think that _just_ crosses the line into explaining terraform - this text is just explaining the example in the context of an example  similar to what we're deploying.   

It's also worth noting that I think we want to explicitly and implicitly encourage users to tag, the alternative is dangerous as it makes it nearly impossible to track resources in case of terraform state management failure.

Reply:
OK. We should still update the ref on the template-deploy repo to use the master branch, right?

Reply:
Yeah, thanks for catching those.

Reply:
33be077f4